libs/platform/src/lib/components/table/table.component.ts

```diff
@@ -559,141 +671,138 @@ export class TableComponent<T = any> implements AfterViewInit, OnDestroy, OnChan
-
-        if (!isNaN(aNumber) && !isNaN(bNumber)) {
-            return (aNumber > bNumber ? 1 : -1) * ascModifier;
-        }
-
-        return (a.key > b.key ? 1 : -1) * ascModifier;
-    };
-
     /** @hidden */
     _isColumnPopoverOpened(key: string): boolean {
         return this._popoverOpen && this._popoverColumnKey === key;
     }
 
     /** @hidden */
-    private _getTableColumnsAsArray(): TableColumnComponent[] {
-        return this.columns?.toArray() || [];
-    }
-
-    /** @hidden */
-    private _setFreezableInfo(): void {
-        this._freezableColumns = this._getFreezableColumn();
-        this._selectionColumnsSize = SELECTION_COLUMN_WIDTH.get(`${this.selectionMode}-${this.contentDensity}`) || 0;
-        this._tablePadding = this._selectionColumnsSize + this._columnsSize * (this._freezableColumns?.length || 0);
+    _setPopoverColumnKey(columnKey: string): void {
+        this._popoverColumnKey = columnKey;
     }
 
-    /** @hidden */
-    private _getFreezableColumn(): string[] {
-        const columns = [];
-
-        if (!this.columns?.length || !this.freezeColumnsTo) {
-            return columns;
-        }
-
-        for (const column of this.columns.toArray()) {
-            if (!column.key) {
-                continue;
-            }
-            if (column.key === this.freezeColumnsTo) {
-                columns.push(column.key);
-
-                return columns;
-            }
-
-            columns.push(column.key);
-        }
-
-        return [];
+    /**
+     * @hidden
+     * Expand/Collapse group row
+     */
+    _toggleGroupRow(groupRow: TableRow): void {
+        this._toggleExpandableTableRow(groupRow);
     }
 
     /** @hidden */
     private _setInitialState(): void {
         const prevState = this.getTableState();
+        const columns = this.columns.toArray();
+        const visibleColumns =
+            this.initialVisibleColumns ||
+            (prevState.columns.length ? prevState.columns : columns.map(({ name }) => name));
 
         this.setTableState({
             ...prevState,
-            freezeToColumn: this.freezeColumnsTo || prevState.freezeToColumn,
-            columns: this.columns.toArray().map((column) => column.key)
+            columns: visibleColumns,
+            sortBy: this.initialSortBy || prevState.sortBy,
+            filterBy: this.initialFilterBy || prevState.filterBy,
+            groupBy: this.initialGroupBy || prevState.groupBy,
+            freezeToColumn: this.freezeColumnsTo || prevState.freezeToColumn
         });
     }
 
+    /** @hidden */
+    private _listenToTableRowsPipe(): void {
+        this._subscriptions.add(
+            this._dataSourceItemsSubject
+                .asObservable()
+                .pipe(
+                    // map source items to table rows
+                    switchMap((source: T[]) => {
+                        return of(this._createTableRowsByDataSourceItems(source));
+                    }),
```

Comment:
Could be one-liner:
```suggestion
                    switchMap((source: T[]) => of(this._createTableRowsByDataSourceItems(source))),
```

Reply:
`return statement` is more convinient in order to add debug breakpoint :-)